### DISCUSSION TITLE
Handle missing partition without crash

### DISCUSSION THREAD

**Discussion by davetapley** [dagster-io/dagster#10880] [Q&A] [answered]
If I have:
```py
@asset(partitions_def=StaticPartitionsDefinition(['one', 'two']))
def stuff(context):
```

and I materialize both partitions, then I can access them via a dict in e.g:

```py
@asset
def use_stuff(context, stuff):
  stuff.keys() # ['one', 'two']
```

... so far so nice.

But if I only materialize e.g. the `one` partition, then I fail hard with:

```
dagster._core.errors.DagsterExecutionLoadInputError: Error occurred while loading input "stuff" of step "use_stuff":
The above exception was caused by the following exception:
FileNotFoundError: [Errno 2] No such file or directory: '/home/vscode/dagster_home/storage/stuff/10'
```

... which I makes sense, but it sure would be nice if I just got this instead:

```py
  stuff.keys() # ['one']
```

But as far as I can tell there's not much I can do, since the exception is thrown deep inside Dagster internals?
Ideas welcome! 🙏🏻

**Comment by OwenKephart** [comment, marked as answer]
Hi @davetapley ! This should be better documented, but this is actually possible to workaround, by setting some metadata on the input that tells Dagster to allow missing partitions:

```
from dagster import asset, AssetIn

@asset(ins={"stuff": AssetIn(metadata={"allow_missing_partitions": True})})
def use_stuff(context, stuff):
  stuff.keys()
```

In the general case, you can also define your own custom IOManager that handles partitioned assets in whatever way you choose (the default implementation can be found here: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/storage/upath_io_manager.py?L111)
